content/200-concepts/100-components/02-prisma-client/051-working-with-fields/working-with-json-fields.mdx

```diff
@@ -46,134 +44,746 @@ export declare type JsonValue = string | number | boolean | null | JsonObject |
+    },
+  },
+})
+```
+
+The following array filters are available:
 
-enum Role {
-  ADMIN
-  USER
-  MODERATOR
+- [`array_contains`](../../../../reference/api-reference/prisma-client-reference#array_contains) <span class="api"></span>
+- [`array_starts_with`](../../../../reference/api-reference/prisma-client-reference#array_starts_with) <span class="api"></span>
+- [`array_ends_with`](../../../../reference/api-reference/prisma-client-reference#array_ends_with) <span class="api"></span>
+
+### Filtering on nested array value
+
+You can filter on the presence of a specific value in a scalar array (strings, integers). In the following examples, the value of `extendedPetsData` includes nested scalar arrays of names:
+
+```json
+{
+  "cats": { "owned": ["Bob", "Sunny"], "fostering": ["Fido"] },
+  "dogs": { "owned": ["Ella"], "fostering": ["Prince", "Empress"] }
 }
 ```
 
-</details>
+#### Scalar value arrays
+
+The following query returns all users that foster a cat named `"Fido"`:
+
+<TabbedContent tabs={[<FileWithIcon text="PostgreSQL" icon="database"/>, <FileWithIcon text="MySQL" icon="database"/>]}>
+<tab>
 
 ```ts
-var json = { pets: [{ name: 'Bob the dog' }, { name: 'Claudine the cat' }] }
+const getUsers = await prisma.user.findMany({
+  where: {
+    extendedPetsData: {
+      path: ['cats', 'fostering'],
+      array_contains: 'Fido',
+    },
+  },
+})
+```
 
-const get = await prisma.user.findMany({
+</tab>
+<tab>
+
+```ts
+const getUsers = await prisma.user.findMany({
   where: {
-    extendedProfile: {
-      equals: json,
+    extendedPetsData: {
+      path: '$.cats.fostering',
+      array_contains: 'Fido',
     },
   },
 })
 ```
 
-The following example omits the `equals` but produces the same result:
+</tab>
+</TabbedContent>
+
+#### JSON object arrays
+
+<TabbedContent tabs={[<FileWithIcon text="PostgreSQL" icon="database"/>, <FileWithIcon text="MySQL" icon="database"/>]}>
+<tab>
 
 ```ts
-var json = { pets: [{ name: 'Bob the dog' }, { name: 'Claudine the cat' }] }
+const json = [{ status: 'expired', insuranceID: 92 }]
 
-const get = await prisma.user.findMany({
+const checkJson = await prisma.user.findMany({
   where: {
-    extendedProfile: json,
+    extendedPetsData: {
+      path: ['insurances'],
+      array_contains: json,
+    },
   },
 })
 ```
 
-The following example returns all users where the `extendedProfile` field does **not** match `json`:
+</tab>
+<tab>
 
 ```ts
-var json = { pets: [{ name: 'Bob the dog' }, { name: 'Claudine the cat' }] }
+const json = { status: 'expired', insuranceID: 92 }
 
-const get = await prisma.user.findMany({
+const checkJson = await prisma.user.findMany({
   where: {
-    extendedProfile: {
-      not: json,
+    extendedPetsData: {
+      path: '$.insurances',
+      array_contains: json,
+    },
+  },
+})
+```
+
+</tab>
+</TabbedContent>
+
+- If you are using PostgreSQL, you must pass in an array of objects to match, even if that array only contains one object:
+
+  ```json
```

Comment:
`json` doesn't support comments, so probably better to just annotate this as `js` here:

![image](https://user-images.githubusercontent.com/4058327/120005154-de193580-bfd7-11eb-8d18-6916d860455a.png)


```suggestion
  ```js
```